src/Features/Core/Portable/UseCoalesceExpression/AbstractUseCoalesceExpressionDiagnosticAnalyzer.cs

```diff
@@ -95,12 +97,30 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
             }
 
             if (!syntaxFacts.AreEquivalent(
-                    conditionRightIsNull ? conditionLeftLow : conditionRightLow, 
+                    conditionRightIsNull ? conditionLeftLow : conditionRightLow,
                     isEquals ? whenFalseNodeLow : whenTrueNodeLow))
             {
                 return;
             }
 
+            var semanticModel = context.SemanticModel;
+            var conditionType = semanticModel.GetTypeInfo(
+                conditionLeftIsNull ? conditionRightLow : conditionLeftLow, cancellationTokan).Type;
+            if (conditionType != null &&
```

Comment:
💭 The problem is one of the reasons `conditionType` would be null is if you reference a value type in a namespace that isn't imported yet. Considering that a fix all is provided for this analyzer, it reasonable to ignore erroneous code for now.